jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -45,16 +45,19 @@ public static void premain(String agentArgument, Instrumentation instrumentation
      pool.setName("jmx_exporter");
      server = new Server(pool);
 
-     ServerConnector cnx = new ServerConnector(server);
-     cnx.setReuseAddress(true);
-     cnx.setHost(socket.getHostName());
-     cnx.setPort(port);
+     ServerConnector connector = new ServerConnector(server);
+     connector.setReuseAddress(true);
+     connector.setHost(host);
+     connector.setPort(port);
+     server.setConnectors(new Connector[]{connector});
 
      ServletContextHandler context = new ServletContextHandler();
      context.setContextPath("/");
      context.addFilter(GzipFilter.class, "/*", null);
      server.setHandler(context);
      context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
+
+     server.setStopAtShutdown(true);
```

Comment:
I'm not sure we should be setting this. If a server takes a long time to shutdown this would prevent monitoring it during that time.